python/ray/_private/worker.py

```diff
@@ -553,6 +553,15 @@ def set_debugger_port(self, port):
         ray._private.state.update_worker_debugger_port(worker_id, port)
         self._debugger_port = port
 
+    @contextmanager
+    def task_paused_by_debugger(self):
+        """Use while the task is paused by debugger"""
+        try:
+            self.core_worker.update_task_is_debugger_paused(True)
+            yield
+        finally:
+            self.core_worker.update_task_is_debugger_paused(False)
+
```

Comment:
is this just for testing?

Reply:
this is the api that can be used from other parts of ray by doing:
```
 with ray._private.worker.global_worker.task_paused_by_debugger():
     # setup the debugger and wait for the client
```

Reply:
i see - so this is aimed as a developer API? I guess I am not too sure when "other parts of ray" would be doing this other than during testing. Anything you have in mind?

Reply:
what we have in mind is something similar to `rpdb.py`, you can check the full prototype PR for more info